gslib/boto_translation.py

```diff
@@ -546,6 +561,7 @@ def GetObjectMedia(self,
     if serialization_data:
       serialization_dict = json.loads(serialization_data)
       key = pickle.loads(binascii.a2b_base64(serialization_dict['url']))
+      _AddCustomEndpointToKey(key)
```

Comment:
Will this run for S3 interaction as well? IN that case, will we end up adding the gs_host for s3?

Reply:
Yes, it looks like `BotoTranslation` is used for both S3 and GCS XML.
Added a provider check!